lib/net/modules/dns.toit

```diff
@@ -640,29 +641,30 @@ class SrvResource extends StringResource:
 
 /**
 Decodes a name from a DNS (RFC 1035) packet.
-The block is invoked with the index of the next data in the packet.
+
+Takes both the $reader and the $packet, as the parts can contain
+  pointers that are absolute in the packet.
 */
-decode-name packet/ByteArray position/int [position-block] -> string:
+decode-name reader/io.Reader packet/ByteArray -> string:
   parts := []
-  parts_ packet position parts position-block
+  parts_ reader packet parts
   return parts.join "."
 
-parts_ packet/ByteArray position/int parts/List [position-block] -> none:
-  while packet[position] != 0:
-    size := packet[position]
+parts_ reader/io.Reader packet/ByteArray parts/List -> none:
+  while true:
+    size := reader.peek-byte
+    if size == 0: break
```

Comment:
Why change the code for zero length parts? Now we don't add a zero length string.

Reply:
as discussed in person. This should still do the same thing.